### DISCUSSION TITLE
Workflow for converting 2024 Java units library code to 2025 Java units library code?

### DISCUSSION THREAD

**Discussion by chauser** [wpilibsuite/2025Beta#50] [Q&A]
Can anyone suggest a smooth and efficient workflow, preferably using VS Code, for converting Java code that uses the 2024 units library to the new 2025 units library? I have literally hundreds of errors after importing the old code to 2025. A recipe for adjusting imports and changing uses in the code itself might save a lot of time for teams in our situation. It's never quite obvious to me what VS code has in the way of global change capabilities that can prevent having to look at and change every place where there's an error.

**Comment by sciencewhiz** [comment]
Was it still hundreds of errors when imported with beta 2? The beta 2 importer should have updated the imports and some of the simpler pieces of code. If you find things that can be fixed by find and replace, please open issues.

**Reply by chauser** [reply]
The imports that are wrong in our code after importing the project with beta 2 are:
`import static edu.wpi.first.units.measure.Units.*;` -- should be edu.wpi.first.units.Units.* -- and
`import edu.wpi.first.units.measure.Measure;` -- should be edu.wpi.first.units.Measure
Global replace on those fixed 150 problems. It appears this is a matter of overzealous conversion rather than missed conversion as the correct imports are what was in the original code.
I'll open an issue.
